### ISSUE TITLE
Multiple Asyncify.handleSleep with WASM frontend does not work

### ISSUE THREAD

**Issue by Srinirajini** [emscripten-core/emsdk#344]
Hi All,
  I am just checking a asynchronous code execution in WebAssembly using a simple timer following the tutorial in this link "https://emscripten.org/docs/porting/asyncify.html" under the section "Returning values" in this link. My code is as simple as creating two Async API and invoke those from main process. I am running in to the "RuntimeError: unreachable" Error. Also noticed that the code is keep going in to the loop because I could not able to see the log "**Returning ....**" which is at the last line of my main process.

> Uncaught (in promise) **RuntimeError: unreachable**
>     at wasm-function[26]:175
>     at Object.ret.(anonymous function) [as j] (http://localhost:8080/async.js:1:36469)
>     at http://localhost:8080/async.js:1:37689
>     at Promise.then (http://localhost:8080/async.js:1:10927)
> async.js:1 Timer expired start_timer1() --->
>  Uncaught (in promise) **RuntimeError: unreachable**
>     at wasm-function[26]:175
>     at Object.ret.(anonymous function) [as j] (http://localhost:8080/async.js:1:36469)
>     at http://localhost:8080/async.js:1:37689
>     at Promise.then (http://localhost:8080/async.js:1:11122)

below is my code snippet.

> ```#include <emscripten.h>
> #include <stdio.h>
> EM_JS(void, start_timer, (), {
>         Asyncify.handleSleep(wakeUp => {
>                 new Promise((resolve, reject) => {
>                         setTimeout(() => {
>                                 resolve();
>                         }, 2500);
>                 })
>                 .then(() => {
>                         console.log('Timer expired start_timer() --->');
>                         wakeUp(1);
>                 })
>         });
> })
> EM_JS(void, start_timer1, (), {
>         Asyncify.handleSleep(wakeUp => {
>                 new Promise((resolve, reject) => {
>                         setTimeout(() => {
>                                 resolve();
>                         }, 3000);
>                 })
>                 .then(() => {
>                         console.log('Timer expired start_timer1() --->');
>                         wakeUp(1);
>                 })
>         });
> })
> int main() {
>   start_timer();
>   printf("Exiting ....\n");
>   start_timer1();
>   printf("Returning ....\n");
>   return 0;
> }```
The command I use to build this code is 

> **emcc asyncTimer.c -O3 -s ASYNCIFY -s 'ASYNCIFY_IMPORTS=["start_timer, start_timer1"]' -s WASM=1 -o async.html**

doesn't know how the compiler treats this code. If anybody could point me the reason for the error "RuntimeError: unreachable" would be great.
appreciate your response.

Thanks,
Srini.

**Comment by kripken** [comment]
Is this related to #329?

**Comment by kripken** [comment]
Looking into this one, the error is in `-s 'ASYNCIFY_IMPORTS=["start_timer, start_timer1"]' ` - that creates an array with a single string value `start_timer, start_timer1` instead of two strings. It works for me with `-s 'ASYNCIFY_IMPORTS=["start_timer","start_timer1"]'`.

While looking at this I realized we don't have good error messages for a mistake like this - sorry about that! I opened #9358 now which would have helped.
